src/session/refreshStoredAppSession.ts

```diff
@@ -0,0 +1,54 @@
+import { AppSession } from './types'
+import { shouldRefresh } from './shouldRefresh/shouldRefresh'
+import { refreshAppSession } from './refreshAppSession/refreshAppSession'
+import { refreshToken } from '../storyblok-auth-api/refreshToken'
+import { putSession, removeSession } from './crud'
+import { AuthHandlerParams } from '../storyblok-auth-api'
+import { GetCookie, SetCookie } from '../utils'
+
+export type RefreshParams = Pick<
+  AuthHandlerParams,
+  'clientSecret' | 'clientId' | 'baseUrl' | 'endpointPrefix'
+>
+export type Refresh = (
+  params: RefreshParams,
+  getCookie: GetCookie,
+  setCookie: SetCookie,
+  appSession: AppSession | undefined,
+) => Promise<AppSession | undefined>
+
+/**
+ * Given a stored session and getters and setters for mutating the storage, refreshes the session
+ * @param params
+ * @param getCookie
+ * @param setCookie
+ * @param currentAppSession
+ */
+export const refreshStoredAppSession: Refresh = async (
+  params,
+  getCookie,
+  setCookie,
+  currentAppSession,
+) => {
+  if (!currentAppSession) {
+    // passed undefined
+    return undefined
+  }
+  if (!shouldRefresh(currentAppSession)) {
+    // does not need refresh
+    return currentAppSession
+  }
+
+  // should refresh
+  const newAppSession = await refreshAppSession(refreshToken(fetch)(params))(
+    currentAppSession,
+  )
```

Comment:
Is this always called from the browser? Or is there a possibility that this can be run on the server side?

Reply:
ah okay. then don't we need to pass `fetch` from `node-fetch` package?

Reply:
Gosh, you're right 🤯 

In the very beginning, the library used to have code running in the client as well as on the server. It seems like I've forgotten to purge the tsconfig `compilerOptions.lib` `"dom"` option.

The library has only been used by us in a Next which uses the isomorphic-fetch, so we never noticed.

I've opened a new ticket for it here: https://storyblok.atlassian.net/browse/EXT-1531

Reply:
Nice! Does the ticket include the replacement of fetch with node-fetch? Or how do you want to approach that part?